src/data-access/user.js

```diff
@@ -0,0 +1,88 @@
+import { Op } from 'sequelize';
+import sequelize from '../db/db.js';
+import User from '../models/user.js';
+import DataAccessError from '../errors/errors.js';
+
+async function createUser(login, password, age) {
+  try {
+    const user = await User.create({ login, password, age });
+
+    return user;
+  } catch (err) {
+    throw new DataAccessError();
+  }
+}
+
+async function updateUser(id, login, password, age) {
+  try {
+    const user = await User.update(
+      { login, password, age },
+      {
+        where: {
+          id,
+        },
+      },
+    );
+
+    return user[0];
+  } catch (err) {
+    throw new DataAccessError();
+  }
+}
+
+async function deleteUser(id) {
+  try {
+    const user = await User.update(
+      { isDeleted: true },
+      {
+        where: {
+          id,
+        },
+      },
+    );
+
+    return user[0];
+  } catch (err) {
+    throw new DataAccessError();
+  }
+}
+
+async function findUserById(id, showDeleted = false) {
+  try {
+    const user = await User.findOne({
+      where: {
+        id,
+        isDeleted: {
+          [Op.eq]: showDeleted,
+        },
+      },
+    });
+
+    return user;
+  } catch (err) {
+    throw new DataAccessError();
+  }
+}
+
+async function findUsersByLogin(loginSubstring, limit) {
+  try {
+    const users = await User.findAll({
+      where: {
+        isDeleted: {
+          [Op.eq]: false,
+        },
+        login: {
+          [Op.like]: `${loginSubstring}%`,
+        },
+      },
+      order: sequelize.col('login'),
+      limit: limit || 0,
```

Comment:
if limit is 0 there is no sense to make db call, as it will always return an empty array

Reply:
Fixed